src/mapping/RadialGeoMultiscaleMapping.cpp

```diff
@@ -0,0 +1,217 @@
+      for (size_t i = 0; i < (outSize - 1); i++) {
+        auto axisPositionCurrent = output()->vertices()[i].getCoords()[effectiveCoordinate];
+        auto axisPositionNext    = output()->vertices()[i + 1].getCoords()[effectiveCoordinate];
+        axisMidpoints(i)         = (axisPositionCurrent + axisPositionNext) / 2;
+      }
+      axisMidpoints(outSize - 1) = std::numeric_limits<double>::max(); // large number, such that vertices after the last midpoint are still assigned
+
+      std::vector<int> counters(outSize); // counts number of vertices in between midpoints for averaging
+
+      // Identify which vertex (index) of the 3D mesh corresponds to which vertex (index) of the 1D meesh
+      _vertexIndicesCollect.clear();
+      _vertexIndicesCollect.reserve(input()->vertices().size());
+      for (size_t i = 0; i < inSize; i++) {
+        auto vertexCoords = input()->vertices()[i].getCoords()[effectiveCoordinate];
+        int  index        = 0;
+        while (vertexCoords > axisMidpoints(index)) {
+          PRECICE_ASSERT(index + 1 < static_cast<int>(outSize));
+          index++;
+        }
+        _vertexIndicesCollect.push_back(index);
+        counters[index] += 1;
+      }
+      _vertexCounter = std::move(counters);
+    }
+
+  } else {
+    PRECICE_ASSERT(getConstraint() == CONSERVATIVE);
+    PRECICE_ASSERT(false, "Not yet implemented");
+    PRECICE_DEBUG("Compute conservative mapping");
+  }
+  _hasComputedMapping = true;
+}
+
+void RadialGeoMultiscaleMapping::clear()
+{
+  PRECICE_TRACE();
+  _vertexIndicesSpread.clear();
+  _vertexIndicesCollect.clear();
+  _hasComputedMapping = false;
+}
+
+void RadialGeoMultiscaleMapping::mapConservative(const time::Sample &inData, Eigen::VectorXd &outData)
+{
+  PRECICE_ASSERT(getConstraint() == CONSERVATIVE);
+  PRECICE_ASSERT(false, "Not yet implemented");
+  PRECICE_DEBUG("Map conservative");
+}
+
+void RadialGeoMultiscaleMapping::mapConsistent(const time::Sample &inData, Eigen::VectorXd &outData)
+{
+  PRECICE_TRACE();
+
+  const int              inDataDimensions = inData.dataDims;
+  const Eigen::VectorXd &inputValues      = inData.values;
+  Eigen::VectorXd &      outputValues     = outData;
+
+  size_t const inSize  = input()->vertices().size();
+  size_t const outSize = output()->vertices().size();
+
+  PRECICE_ASSERT((inputValues.size() / inDataDimensions == static_cast<int>(input()->vertices().size())),
+                 inputValues.size(), inDataDimensions, input()->vertices().size());
+  PRECICE_ASSERT((outputValues.size() / inDataDimensions == static_cast<int>(output()->vertices().size())),
+                 outputValues.size(), inDataDimensions, output()->vertices().size());
+
+  PRECICE_DEBUG("Map consistent");
+  if (_type == MultiscaleType::SPREAD) {
+    // assign 1D vertex value to all 3D vertices in vicinity
+    for (size_t i = 0; i < outSize; i++) {
+      outputValues((i * inDataDimensions)) = inputValues(_vertexIndicesSpread[i]);
```

Comment:
`outputValues((i * inDataDimensions)) =` always assigns the first coordinate of each output data for each vertex. Assuming the output values has the same dimensionality as the input values.

`inputValues(_vertexIndicesSpread[i]);` gets the value for the assigned vertex, which only works if `inDataDimensions == 1`.

Reply:
> Assuming the output values has the same dimensionality as the input values.

We now have an assertion for that.

> `inputValues(_vertexIndicesSpread[i]);` gets the value for the assigned vertex, which only works if `inDataDimensions == 1`.

The `_vertexIndicesSpread` refers to a concrete vertex. Since this is currently always 3D, we should extract only the first component as well.

What is happening now?
- Only one component of `outputValues` is set, using only the first component of `inputValues` and discarding the rest, or
- All components of `outputValues` are set equal to all components of `inputValues`, and the rest of the components happen to be zero?

Reply:
The `_vertexIndicesSpread` indeed correspond to the vertices, not to the values. So, we still need to multiply by `inDataDimensions` to access th correct input value.

As for the `outputValues((i * inDataDimensions))`, I see what you meant. I have now changed all such occurences to `outDataDimensions`.